host-bin/enter-chroot

```diff
@@ -385,10 +385,139 @@ if ! mountpoint -q "$destmedia"; then
+downloads ~/Downloads
+EOF
+fi
+# Bind-mount the stuff in $CHROOT/etc/crouton/shares, unless NOLOGIN is set
+if [ -z "$NOLOGIN" -a -f "$shares" ]; then
+    localdownloads='/home/chronos/user/Downloads'
+    if [ ! -d "$localdownloads" ]; then
+        localdownloads=''
+    fi
+    localencrypted='/home/chronos/user/crouton/shared'
+    if [ ! -d "/home/chronos/user" ]; then
+        localencrypted=''
+    fi
+    localshare='/mnt/stateful_partition/crouton/shared'
+    # Parse file and split into three lines: src, dest, options
+    awk '
+        BEGIN {
+            PARAMMATCH = "^[ \t]*(\"[^\"]+\"|[^ \t\"]+)([ \t]"
+        }
+        function strip(s) {
+            if (index(s, "\"")) {
+                sub("[ \t]*\"", "", s)
+                sub("\"[ \t]*", "", s)
+            } else {
+                gsub("[ \t]", "", s)
+            }
+            return s
+        }
+        $1 ~ /^#/ { next }
+        match($0, PARAMMATCH ")") {
+            dest = substr($0, RLENGTH+1)
+            src = strip(substr($0, 1, RLENGTH))
+            if (match(dest, PARAMMATCH "|$)")) {
+                opts = strip(substr(dest, RLENGTH+1))
+                dest = strip(substr(dest, 1, RLENGTH))
+                if (length(src) && length(dest) && !match(opts, "[^a-z,]")) {
+                    # Ensure that they end in slashes
+                    if (substr(src, length(src)) != "/")
+                        src = src "/"
+                    if (substr(dest, length(dest)) != "/")
+                        dest = dest "/"
+                    # .. path elements are not allowed
+                    if (!index(src, "/../") && !index(dest, "/../")) {
+                        print src
+                        print dest
+                        print opts
+                        next
+                    }
+                }
+            }
+        }
+        /./ {
+            print "Invalid syntax in /etc/crouton/shares on line " NR ":\n  " \
+                  $0 > "/dev/stderr"
+        }
+    ' "$shares" | while read -r src && read -r dest && read -r opts; do
+        line="\n  \"$src\" \"$dest\" $opts"
+        # Expand src
+        case "${src%%/*}" in
+            download|downloads)
+                if [ -z "$localdownloads" ]; then
+                    echo "Not mounting share (no Chromium OS user):$line" 1>&2
+                    continue
+                fi
+                src="$localdownloads/${src#*/}";;
+            encrypt|encrypted)
+                if [ -z "$localencrypted" ]; then
+                    echo "Not mounting share (no Chromium OS user):$line" 1>&2
+                    continue
+                fi
+                src="$localencrypted/${src#*/}";;
+            share|shares|shared)
+                src="$localshare/${src#*/}";;
+            *)
+                echo "Invalid share:$line" 1>&2
+                continue;;
+        esac
+        # Expand dest for homedirs
+        if [ "${dest#[~/]}" = "$dest" ]; then
```

Comment:
You do not document this case, nor provide an example for it, in the description above.

Either do, or fail in that case (i.e. do not allow path that do not start in `/` or `~`).